### PR TITLE
Created UI that adds a new item to the shopping list and records the …

### PR DESCRIPTION
**Pull request by michhub-dev**
…time of next purchase

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description
 Created a form that allows a user to add an item to their shopping list and select the next probable time of purchase. The user gets a message when the item is successfully added to the database. 

## Acceptance Criteria

UI-related tasks:
- [x] The `AddItem` view displays a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again. There should be 3 choices for this:
	- “Soon”, corresponding to 7 days
	- “Kind of soon”, corresponding to 14 days
	- “Not soon”, corresponding to 30 days
- [x] The input that accepts the name of the item has a semantic `label` element associated with it
- [x] The user can submit this form with both the mouse and the `Enter` key  
- [x] When the user submits the form, they see a message indicating that the item either was or was not saved to the database.

Data-related tasks:
- [x] The `console.log` in the `addItem` function in `src/api/firebase.js`  is replaced with a function that adds the new document to the Firestore database. That function will be imported from the `firebase/firestore` module.
- [x]  The user’s soon/not soon/kind of soon choice is used to calculate `nextPurchasedDate`
## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Testing Steps / QA Criteria
From your terminal, pull down this branch with git pull origin `git pull origin md-vo-add-new-item-to-list` and check that branch out with
`git checkout md-vo-add-new-item-to-list` 
From the UI add an item and select a duration. When the item button is clicked the user will see a message indicating if the item was added to the database. If successful, the item should be added on the list view.

### CODE REVIEW

File: src/views/AddItem.jsx [hunk 1]
```diff
@@ -1,3 +1,77 @@
...
+								defaultChecked
+							/>
+							<label htmlFor="soon">Soon</label>
+						</div>
+
+						<div>
+							<input
+								type="radio"
+								id="kind of soon"
+								name="duration"
```
Comment (klezi10) [inline src/views/AddItem.jsx]: Not to be nitpicky but I believe best practices for id value is with dashes like "kind-of-soon"

**Review by klezi10 — APPROVED** [review]
Well done! I see the item is saved to firebase and it shows up in the List page.

**Review by mide358 — APPROVED** [review]
Nicely done!

File: src/views/AddItem.jsx (hunk 1 above)
Comment (michhub-dev) [reply src/views/AddItem.jsx]: > Not to be nitpicky but I believe best practices for id value is with dashes like "kind-of-soon"

Thank you for bringing that to our attention. Fixed now

File: src/views/AddItem.jsx [hunk 2]
```diff
@@ -1,3 +1,77 @@
...
+
+		let listId = 'my test list';
+
+		const item = await addItem(listId, {
+			itemName: newItem,
+			daysUntilNextPurchase: nextPurchaseTime,
+		});
+
+		if (item.id) {
+			setStatusMessage('Item Successfully Added!');
```
Comment (redapy) [inline src/views/AddItem.jsx]: once you add an item, the success message is always present in the UI. Maybe you can add a timeout to make the message disappear after some seconds.

File: src/views/AddItem.jsx [hunk 3]
```diff
@@ -1,3 +1,77 @@
...
+					<fieldset onChange={(e) => setPurchaseTime(e.target.value)}>
+						<legend>How soon will you buy this again?</legend>
+
+						<div>
+							<input
+								type="radio"
+								id="soon"
+								name="duration"
+								value={7}
+								defaultChecked
```
Comment (redapy) [inline src/views/AddItem.jsx]: example of a controlled radio button: 
```
onChange={(e) => setPurchaseTime(e.target.value)}
checked={7 === parseInt(nextPurchaseTime)}
```
and just remove the `onChange` function from the `fieldset`.
What we are doing here, is just deciding if a radio button is checked or not based on its value (7 for this input) and our current state.

**Review by redapy — CHANGES_REQUESTED** [review]
Great work, clean code, and perfect consideration for a11y! I left some small notes that affect a bit the UI.
After adding an item, the `nextPurchaseTime` got updated to the default value (7), which is great, but that change is not reflected on the UI.  That's because you are using uncontrolled radio buttons, you can do a small search about controlled VS uncontrolled inputs.

File: src/views/AddItem.jsx [hunk 4]
```diff
@@ -5,25 +5,34 @@ export function AddItem() {
...
-			setStatusMessage('Item Successfully Added!');
+			setAlertVisible(true);
+			setStatusMessage(`${newItem} Successfully Added!`);
+
 			setNewItem('');
 			setPurchaseTime(7);
+
+			setTimeout(() => {
+				setAlertVisible(false);
+			}, 3000);
```
Comment (redapy) [inline src/views/AddItem.jsx]: you can clear it directly here without having an additional state. just replace `setAlertVisible(false)` with `setStatusMessage("")`

File: src/views/AddItem.jsx [hunk 5]
```diff
@@ -59,18 +70,25 @@ export function AddItem() {
...
 
 						<div>
-							<input type="radio" id="not_soon" name="duration" value={30} />
+							<input
+								type="radio"
+								id="not_soon"
+								name="duration"
+								value={30}
+								onChange={(e) => setPurchaseTime(e.target.value)}
+							/>
```
Comment (redapy) [inline src/views/AddItem.jsx]: you need to add the `checked` attribute for each input as I mentioned in my example.

**Review by redapy — CHANGES_REQUESTED** [review]

**Review by redapy — APPROVED** [review]
all working as expected, great work team!

**Comment by michhub-dev** [comment]
> all working as expected, great work team!

Thank you for the review and feedback!

**Pull request merged**